common/numberfunctions.lua

```diff
@@ -212,4 +212,58 @@ if not math.HSLtoRGB then
 
 		return cr, cg, cb
 	end
+
+
+	if not math.distance2dSquared then
+		function math.distance2dSquared(x1, z1, x2, z2)
+			local x = x1 - x2
+			local z = z1 - z2
+			return x * x + z * z
+		end
+	end
+
+	if not math.distance2d then
+		function math.distance2d(x1, z1, x2, z2)
+			return math.sqrt(math.distance2dSquared(x1, z1, x2, z2))
```

Comment:
```suggestion
			return math.hypot(x1-x2, z1-z2)
```

Reply:
What's the advantage of this?
Based on https://springrts.com/wiki/Lua_MathExtra this seems like a questionable change

Reply:
mostly for being idiomatic. if you're worried about speed then you can also do `math.diag`
```suggestion
			return math.diag(x1-x2, z1-z2)
```